support-data-rest/src/main/java/org/uniprot/api/configure/uniprot/domain/model/AdvanceUniProtKBSearchTerm.java

```diff
@@ -0,0 +1,148 @@
+package org.uniprot.api.configure.uniprot.domain.model;
+
+import java.io.Serializable;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.stream.Collectors;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+import org.apache.commons.lang3.StringUtils;
+import org.uniprot.store.config.searchfield.common.SearchFieldConfig;
+import org.uniprot.store.config.searchfield.factory.SearchFieldConfigFactory;
+import org.uniprot.store.config.searchfield.factory.UniProtDataType;
+import org.uniprot.store.config.searchfield.model.SearchFieldItem;
+import org.uniprot.store.config.searchfield.model.SearchFieldType;
+import org.uniprot.store.search.domain.EvidenceGroup;
+import org.uniprot.store.search.domain.impl.AnnotationEvidences;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+@Data
+@Builder
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class AdvanceUniProtKBSearchTerm implements Serializable {
+    private static final long serialVersionUID = -5776203445383103470L;
+    private String id;
+    @JsonIgnore private String parentId;
+    @JsonIgnore private Integer childNumber;
+    @JsonIgnore private Integer seqNumber;
+    private String label;
+    private String itemType;
+    private String term;
+    private String dataType;
+    private String fieldType;
+    @JsonIgnore private String description;
+    private String example;
+    private String autoComplete;
+    private String autoCompleteQueryTerm;
+    @JsonIgnore private String autoCompleteQueryFieldValidRegex;
+    private String regex;
+    private List<Value> values;
+    private List<AdvanceUniProtKBSearchTerm> items;
+    private List<EvidenceGroup> evidenceGroups;
+
+    @Data
+    @AllArgsConstructor
+    public static class Value {
+        private String name;
+        private String value;
+    }
+
+    public static List<AdvanceUniProtKBSearchTerm> getUniProtKBSearchTerms() {
```

Comment:
This method is called whenever someone goes to the /search_terms in the configure controller. Does it mean multiple calls to `getUniProtKBSearchTerms` will cause repeated navigation of the object structure? If so, can we store the return result in a field, then return it if non-null next time? (badly explained, but hopefully this is clearer:)

```java
private static List<AdvanceUniProtKBSearchTerm> rootSearchTermConfigs;
public static List<AdvanceUniProtKBSearchTerm> getUniProtKBSearchTerms() {
    if(rootSearchTermConfigs == null) {
        // set rootSearchTermConfigs
    }
    return rootSearchTermConfigs;  
}
```